client/src/views/ActiveUsersPage.vue

```diff
@@ -79,36 +172,153 @@ import {
   IonPage,
   IonLabel,
   IonListHeader,
+  IonCheckbox,
+  IonText,
 } from '@ionic/vue';
+import {
+  personRemove,
+  personAdd,
+  eye,
+} from 'ionicons/icons';
 import UserListItem from '@/components/Users/UserListItem.vue';
 import UserCard from '@/components/Users/UserCard.vue';
 import ButtonOption from '@/components/ButtonOption.vue';
 import { isAdmin } from '@/common/permissions';
 import ListGridToggle from '@/components/ListGridToggle.vue';
 import { DisplayState } from '@/components/ListGridToggle.vue';
+import { popoverController } from '@ionic/vue';
+import UserContextMenu from '@/components/UserContextMenu.vue';
+import { UserAction } from '@/components/UserContextMenu.vue';
 import ActionBar from '@/components/ActionBar.vue';
+import { MockUser, getMockUsers } from '@/common/mocks';
+import { Ref } from 'vue';
+import { onMounted } from '@vue/runtime-core';
 
 const displayView = ref(DisplayState.List);
-const userList = ['User1', 'User2', 'User3'];
+const userList: Ref<MockUser[]> = ref([]);
+const userListItemRefs: Ref<typeof UserListItem[]> = ref([]);
+const userGridItemRefs: Ref<typeof UserCard[]> = ref([]);
+const allUsersSelected = ref(false);
+const interminateState = ref(false);
```

Comment:
I'm not sure what this variable does. It's used to set the prop `indeterminate` on one checkbox, and the non-existing `interminate` prop on user-list-item. What's the point of this variable?

Reply:
I wanted to update the state of the "global" checkbox to make it indeterminate if the user selects an item (a user in this case) from the user list, so I used a ref.